pkg/operator/sync.go

```diff
@@ -249,15 +249,15 @@ func newDeployment(config *OperatorConfig, features map[string]bool) *appsv1.Dep
 			},
 			Labels: map[string]string{
 				"api":     "clusterapi",
-				"k8s-app": "controller",
```

Comment:
why do we need this change? can we please have a separate PR for this if needed?

Reply:
I'm trying to use same generated TLC certificates by existing `machine-api-operator` service. Those are used in the existing `kube-rbac-proxy` container, and for that reason I need the label selector on the `serviceMonitor` to match `machine-api-operator` pods for controller too.

Reply:
This would cause issues, the labels need to be distinct per deployment, otherwise we would have the mapi operator and mapi controller deployments believing they both own the same pods. We should generate a new certificate